components/BarcodeScanner.vue

```diff
@@ -0,0 +1,334 @@
+      flat
+      fab
+      :ripple="false"
+      size="16px"
+      color="white"
+      icon="fas fa-times"
+      class="q-ma-sm close-btn"
+      @click.native="close()" />
+    <!-- Buttons -->
+    <Buttons
+      :camera-list="cameraList"
+      @updateCamera="updateCamera($event)"
+      @upload="upload($event)" />
+  </div>
+</template>
+
+<script>
+/*!
+  * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+  */
+import DBR from '../helpers/dbr.js';
+import driverLicenseFields from '../helpers/driverLicenseFields.js';
+import Buttons from './Buttons.vue';
+import Spinner from './Spinner.vue';
+
+DBR.BarcodeReader.license = '100902848';
+
+export default {
+  name: 'BarcodeScanner',
+  components: {
+    Buttons,
+    Spinner
+  },
+  props: {
+    tipText: {
+      type: String,
+      default: 'Scan barcode on the back of license.'
+    },
+    guideColor: {
+      type: String,
+      default: 'rgb(254, 142, 20)'
+    }
+  },
+  data() {
+    return {
+      loadingCamera: true,
+      scanning: false,
+      isDestroyed: false,
+      scanner: null,
+      reader: null,
+      dlInfo: null,
+      clientHeight: null,
+      clientWidth: null,
+      regionScale: 0.4,
+      cameraList: [],
+      currentCamera: null,
+      maskCanvas: null,
+      maskCanvasContext: null,
+    };
+  },
+  computed: {
+    regionMaskEdgeLength() {
+      const regionMaskEdgeLength =
+        this.regionScale * Math.min(this.clientWidth, this.clientHeight);
+      return Math.floor(regionMaskEdgeLength);
+    },
+    regionLeft() {
+      const left =
+        (this.clientWidth - this.regionMaskEdgeLength) / 2 / this.clientWidth;
+      if(this.clientWidth > this.clientHeight) {
+        return Math.round(left * 100) - 25;
+      } else {
+        return Math.round(left * 100) - 20;
+      }
+    },
+    regionTop() {
+      const top =
+        (this.clientHeight - this.regionMaskEdgeLength) / 2 / this.clientHeight;
+      return Math.round(top * 100) - 5;
+    },
+    region() {
+      return {
+        regionLeft: this.regionLeft,
+        regionRight: 100 - this.regionLeft,
+        regionTop: this.regionTop,
+        regionBottom: 100 - this.regionTop,
+        regionMeasuredByPercentage: 1,
+      };
+    },
+  },
+  async mounted() {
+    this.clientHeight = document.body.clientHeight;
+    this.clientWidth = document.body.clientWidth;
+    this.maskCanvas = document.getElementById('guide');
+    this.maskCanvasContext = this.maskCanvas.getContext('2d');
+    this.cvsDrawArea();
+    window.onresize = () => {
+      this.isResizing = true;
+      this.clientHeight = document.body.clientHeight;
+      this.clientWidth = document.body.clientWidth;
+      this.cvsDrawArea();
+      this.isResizing = false;
+    };
+    await this.showScanner();
+    try {
+      this.cameraList = await this.scanner.getAllCameras();
+      this.currentCamera = await this.scanner.getCurrentCamera();
+    } catch(e) {
+      console.log(e);
```

Comment:
Should this be `console.error`?  Does it also need to throw or emit the error?  If, say, a user denies camera access, does that cause an error that needs to be handled here or do those apis just return empy arrays or similar?